contracts/rewards/src/contract.rs

```diff
@@ -26,21 +29,90 @@ pub fn instantiate(
         },
     )?;
 
+    PARAMS.save(
+        deps.storage,
+        &StoredParams {
+            params: msg.params,
+            last_updated: Epoch {
+                epoch_num: 0,
+                block_height_started: env.block.height,
+            },
+        },
+    )?;
+
     Ok(Response::new())
 }
 
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn execute(
     deps: DepsMut,
-    _env: Env,
-    _info: MessageInfo,
+    env: Env,
+    info: MessageInfo,
     msg: ExecuteMsg,
 ) -> Result<Response, axelar_wasm_std::ContractError> {
-    let mut _contract = Contract::new(deps);
     match msg {
-        ExecuteMsg::RecordParticipation { .. } => todo!(),
-        ExecuteMsg::AddRewards { .. } => todo!(),
-        ExecuteMsg::DistributeRewards { .. } => todo!(),
-        ExecuteMsg::UpdateParams { .. } => todo!(),
+        ExecuteMsg::RecordParticipation {
+            event_id,
+            worker_address,
+        } => {
+            let worker_address = deps.api.addr_validate(&worker_address)?;
+            Contract::new(deps)
+                .record_participation(
+                    nonempty::String::try_from(event_id)
```

Comment:
Make it `nonempty::String` in the message struct `ExecuteMsg::RecordParticipation` with custom deserializer.

Reply:
Actually, the deserializer is already there https://github.com/axelarnetwork/axelar-amplifier/blob/271d452c6ae92a9d021112fad986a942b55852cb/packages/axelar-wasm-std/src/nonempty/string.rs#L7. Use it.